Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1481,6 +1481,40 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
+
+    if let modifiers = node.modifiers, modifiers.hasError {
+      for modifier in modifiers {
+        if let detail = modifier.detail, detail.detail != .keyword(.set) {
+          let unexpectedBetweenLeftParenAndDetail = detail.unexpectedBetweenLeftParenAndDetail?.tokens(viewMode: .all)
+          let unexpectedBetweenDetailAndRightParen = detail.unexpectedBetweenDetailAndRightParen?.tokens(viewMode: .all)
+          let unexpectedTokens = (unexpectedBetweenLeftParenAndDetail?.map { $0 } ?? []) + (unexpectedBetweenDetailAndRightParen?.map { $0 } ?? [])
+
+          // If there is no unexpected tokens it means we miss a paren or set keyword.
+          // So we just skip the handling here
+          guard !unexpectedTokens.isEmpty else {
+            return .visitChildren
```

Comment:
I think we still want to continue to the handling below (`let missingTries = …`), so we shouldn’t `return` here.

---

To remove the force unwrap below, this check could also be 

```swift
guard let firstUnexpected = unexpected tokens.first else { … }
```